packages/app-store/routing-forms/pages/routing-link/[...appPages].tsx

```diff
@@ -302,9 +303,11 @@ const usePrefilledResponse = (form: Props["form"]) => {
   form.fields?.forEach((field) => {
     const valuesFromQuery = searchParams?.getAll(getFieldIdentifier(field)).filter(Boolean);
     // We only want to keep arrays if the field is a multi-select
+
     const value = valuesFromQuery.length > 1 ? valuesFromQuery : valuesFromQuery[0];
+
     prefillResponse[field.id] = {
-      value: value || "",
+      value: transformResponse({ field, value }),
```

Comment:
Now, routing link also correctly handles, comma separated strings. Earlier, they were prefilled correctly but routing didn't work correctly for comma separated strings